client/state/sites/vouchers/reducer.js

```diff
@@ -28,6 +33,28 @@ import { itemsSchema } from './schema';
 export const items = ( state = {}, action ) => {
 	const { siteId } = action;
 	switch ( action.type ) {
+		case SITE_VOUCHERS_ASSIGN:
+			switch ( action.serviceType ) {
+				case GOOGLE_AD_CREDITS:
+					const currentVouchers = state[ siteId ]
+						? ( state[ siteId ][ GOOGLE_AD_CREDITS ] || [] )
+						: [];
+
+					const vouchers = currentVouchers.splice( 0 );
```

Comment:
I figured :wink:, but didn't want to assume anything. if that's the case, we could simply `concat()` the new element, because that already clones the array…

```js
const vouchers = currentVouchers.concat( action.voucher );

// or with a list spread operator, though I'm not sure we
// have that enabled yet
const vouchers = [ ...currentVouchers, action.voucher ];
```

I'm not sure what the structure of `state[ siteId ][ GOOGLE_AD_CREDITS ]` might be, but if it's `undefined` and not `false` when empty, this could all be done in one swift pass because the list operations are safe by default for empty lists:

```js
import get from 'lodash/get';

// assuming undefined or empty list [] on empty state
const vouchers = get( state, [ siteId, GOOGLE_AD_CREDITS ], [] ).concat( action.voucher );

// assuming empty list [] on empty state
const vouchers = state[ siteId ][ GOOGLE_AD_CREDITS ].concat( action.voucher );
```

I'm a fan of **lodash**'s `get()` because it will fail softly if the nested path doesn't exist. Direct access of the properties could throw a `TypeException` if the parent property (in this case, `siteId`) doesn't exist.